UNITTESTS/stubs/hal/CMakeLists.txt

```diff
@@ -0,0 +1,25 @@
+# Copyright (c) 2021 ARM Limited. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+add_library(mbed-stubs-hal)
+
+target_compile_definitions(mbed-stubs-hal
+    PRIVATE
+        DEVICE_PWMOUT
+        DEVICE_WATCHDOG
+        MBED_WDOG_ASSERT=1
+)
+
+target_sources(mbed-stubs-hal
+    PRIVATE
+        pwmout_api_stub.c
+        us_ticker_stub.cpp
+        watchdog_api_stub.c
+)
+
+target_link_libraries(mbed-stubs-hal
+    PRIVATE
+        mbed-headers
+        mbed-stubs-headers
+        gcov
```

Comment:
This won't work with AppleClang. We need to pass `target_link_options(mbed-stubs-hal --coverage)` instead. That will alias to the correct linker argument on gcc and AppleClang. The current setup causes a build failure with the error `library not found for -lgcov` when building with AppleClang.